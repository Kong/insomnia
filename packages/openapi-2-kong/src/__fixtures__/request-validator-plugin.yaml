openapi: 3.0.2

info:
  title: Example
  version: 1.0.0

servers:
  - url: http://backend.com/path

x-kong-plugin-request-validator:

paths:
  /global:
    get:
      requestBody:
        content:
          application/json:
            schema:
              $ref: '#/components/schemas/jsonSchema'
  /params/{pathid}/:
    get:
      x-kong-plugin-request-validator:
        enabled: true
        config:
          body_scheme: '{}'
      parameters:
        - in: query
          name: queryid
          schema:
            type: integer
          required: true
        - in: header
          name: User-Id
          schema:
            type: integer
          required: true
        - in: cookie
          name: cookieid
          schema:
            type: integer
          required: true
        - in: path
          name: pathid
          schema:
            type: integer
          required: true
  /body:
    post:
      requestBody:
        content:
          application/json:
            schema:
              $ref: '#/components/schemas/jsonSchema'
          application/xml:
            schema:
              $ref: '#/components/schemas/xmlSchema'

components:
  schemas:
    jsonSchema:
      type: object
      properties:
        id:
          type: integer
        name:
          type: string
    xmlSchema:
      type: object
      properties:
        prop:
          type: integer
