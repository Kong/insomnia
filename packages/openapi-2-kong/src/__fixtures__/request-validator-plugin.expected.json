{
  "_format_version": "1.1",
  "services": [
    {
      "protocol": "http",
      "host": "Example",
      "port": 80,
      "path": "/path",
      "name": "Example",
      "plugins": [
        {
          "config": {
            "body_schema": "{}",
            "version": "draft4"
          },
          "enabled": true,
          "name": "request-validator"
        }
      ],
      "routes": [
        {
          "methods": [
            "POST"
          ],
          "name": "Example-body-post",
          "paths": [
            "/body$"
          ],
          "plugins": [
            {
              "config": {
                "allowed_content_types": [
                  "application/json",
                  "application/xml"
                ],
                "body_schema": "{\"type\":\"object\",\"properties\":{\"id\":{\"type\":\"integer\"},\"name\":{\"type\":\"string\"}}}",
                "version": "draft4"
              },
              "enabled": true,
              "name": "request-validator"
            }
          ],
          "strip_path": false,
          "tags": [
            "OAS3_import",
            "OAS3file_request-validator-plugin.yaml"
          ]
        },
        {
          "methods": [
            "GET"
          ],
          "name": "Example-global-get",
          "paths": [
            "/global$"
          ],
          "plugins": [
            {
              "config": {
                "allowed_content_types": [
                  "application/json"
                ],
                "body_schema": "{\"type\":\"object\",\"properties\":{\"id\":{\"type\":\"integer\"},\"name\":{\"type\":\"string\"}}}",
                "version": "draft4"
              },
              "enabled": true,
              "name": "request-validator"
            }
          ],
          "strip_path": false,
          "tags": [
            "OAS3_import",
            "OAS3file_request-validator-plugin.yaml"
          ]
        },
        {
          "methods": [
            "GET"
          ],
          "name": "Example-params-get",
          "paths": [
            "/params$"
          ],
          "plugins": [
            {
              "config": {
                "parameter_schema": [
                  {
                    "explode": false,
                    "in": "path",
                    "name": "userId",
                    "required": true,
                    "schema": "{\"type\":\"integer\"}",
                    "style": "form"
                  }
                ],
                "version": "draft4"
              },
              "enabled": true,
              "name": "request-validator"
            }
          ],
          "strip_path": false,
          "tags": [
            "OAS3_import",
            "OAS3file_request-validator-plugin.yaml"
          ]
        }
      ],
      "tags": [
        "OAS3_import",
        "OAS3file_request-validator-plugin.yaml"
      ]
    }
  ],
  "upstreams": [
    {
      "name": "Example",
      "tags": [
        "OAS3_import",
        "OAS3file_request-validator-plugin.yaml"
      ],
      "targets": [
        {
          "target": "backend.com:80"
        }
      ]
    }
  ]
}
